providers/hetznercloud/hcapi/api.go

```diff
@@ -0,0 +1,132 @@
+package hcapi
+
+import (
+	"github.com/hetznercloud/hcloud-go/v2/hcloud"
+)
+
+type Client interface {
+	Firewall() FirewallClient
+	Image() ImageClient
+	Network() NetworkClient
+	Server() ServerClient
+	ServerType() ServerTypeClient
+	SSHKey() SSHKeyClient
+}
+
+type client struct {
+	client *hcloud.Client
+}
+
+type ServerTypeClient interface {
+	hcloud.IServerTypeClient
+}
+
+type serverTypeClient struct {
+	hcloud.IServerTypeClient
+}
+
+type ImageClient interface {
+	hcloud.IImageClient
+}
+
+type imageClient struct {
+	hcloud.IImageClient
+}
+
+type SSHKeyClient interface {
+	hcloud.ISSHKeyClient
+}
+
+type sshKeyClient struct {
+	hcloud.ISSHKeyClient
+}
+
+type ServerClient interface {
+	hcloud.IServerClient
+}
+
+type serverClient struct {
+	hcloud.IServerClient
+}
+
+type NetworkClient interface {
+	hcloud.INetworkClient
+}
+
+type networkClient struct {
+	hcloud.INetworkClient
+}
+
+type FirewallClient interface {
+	hcloud.IFirewallClient
+}
+
+type firewallClient struct {
+	hcloud.IFirewallClient
+}
+
+func NewClient(opts ...hcloud.ClientOption) Client {
+	return &client{
+		client: hcloud.NewClient(opts...),
+	}
+}
+
+func NewServerTypeClient(client hcloud.IServerTypeClient) ServerTypeClient {
+	return &serverTypeClient{
+		IServerTypeClient: client,
+	}
+}
+
+func (c *client) ServerType() ServerTypeClient {
+	return NewServerTypeClient(&c.client.ServerType)
```

Comment:
I assume you tried mocking the hetzner client directly without this wrapper without success?

Reply:
Yeah I tried it but failed.

Reply:
Mocking the Hetzner client directly doesn't work because it's not an interface upstream. Only interfaces for the sub-clients exists. If there is another way please let me know.